x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -290,117 +264,72 @@ static String getPolicyId(SnapshotInfo snapshotInfo) {
                 " to have a policy in its metadata, but it did not"));
     }
 
-    /**
-     * Maybe delete the given snapshots. If a snapshot is currently running according to the cluster
-     * state, this waits (using a {@link ClusterStateObserver} until a cluster state with no running
-     * snapshots before executing the blocking
-     * {@link #deleteSnapshots(Map, TimeValue, SnapshotLifecycleStats)} request. At most, we wait
-     * for the maximum allowed deletion time before timing out waiting for a state with no
-     * running snapshots.
-     *
-     * It's possible the task may still run into a SnapshotInProgressException, if a snapshot is
-     * started between the state retrieved here and the actual deletion. Since is is expected to be
-     * a rare case, no special handling is present.
-     */
-    private void maybeDeleteSnapshots(Map<String, List<SnapshotInfo>> snapshotsToDelete,
-                                      TimeValue maximumTime,
-                                      SnapshotLifecycleStats slmStats) {
+    void deleteSnapshots(Map<String, List<SnapshotInfo>> snapshotsToDelete,
+                         SnapshotLifecycleStats slmStats,
+                         ActionListener<Void> listener) {
         int count = snapshotsToDelete.values().stream().mapToInt(List::size).sum();
         if (count == 0) {
             logger.debug("no snapshots are eligible for deletion");
             return;
```

Comment:
This now needs to call the listener `onResponse` handler

Reply:
Right fixed :)